pallets/abc-demo/src/lib.rs

```diff
@@ -301,17 +310,11 @@ decl_module! {
             ) -> dispatch::DispatchResult {
             let sender = ensure_signed(origin)?;
             ensure!(Errands::contains_key(&description_cid), Error::<T>::ErrandTaskNotExist);
-            ensure!(Self::processing_errands_contains(&description_cid),  Error::<T>::ProcessingErrandNotExist);
-            if let Some(errand) = Errands::get(&description_cid) {
-                match T::AccountId::decode(&mut errand.account_id.as_slice()) {
-                    Ok(epr) => {
-                        ensure!(sender == epr, Error::<T>::NoRightToInitErrand);
-                    },
-                    Err(err) => debug::error!("decode account id error: {:?}", err)
-                }
-            } else {
-                debug::error!("errand is not exist")
-            }
+            let sender_account: AccountId32 = Self::account_to_bytes(&sender)?;
+            let accounts: Vec<AccountId32> = vec![sender_account];
+
+            #[cfg(feature = "std")]
+            ensure!(task::account_from_seed_in_accounts("Alice", accounts), Error::<T>::NoRightToUpdateErrand);
```

Comment:
init_errand is called by the "delegate(or sender)" in offchain workers, so there is no need to ensure the available accounts contains Alice.